File: scripts/fetch-openapi.py
import urllib.request
import json

url = "https://lbazkpkvkvarimnqzgqb.supabase.co/rest/v1/"
key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImxiYXprcGt2a3ZhcmltbnF6Z3FiIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3MTI4MzY5MSwiZXhwIjoyMDg2ODU5NjkxfQ.RDYx8KsUzlMNT5qb7WfTF4dCrp3dEx8eqSmC0b1g8E0"

headers = {
    "apikey": key,
    "Authorization": f"Bearer {key}"
}

req = urllib.request.Request(url, headers=headers)
try:
    with urllib.request.urlopen(req) as res:
        data = json.loads(res.read().decode())
        paths = data.get("paths", {})
        rpcs = [path for path in paths if path.startswith("/rpc/")]
        print("Available RPCs:")
        for rpc in rpcs:
            print("  ", rpc)
except Exception as e:
    print("Error:", e)

File: scripts/fetch-openapi.mjs
import dotenv from 'dotenv'
dotenv.config({ path: '.env.local' })

const url = 'https://lbazkpkvkvarimnqzgqb.supabase.co/rest/v1/'
const key = process.env.SUPABASE_SERVICE_ROLE_KEY

async function run() {
    try {
        const response = await fetch(url, {
            headers: {
                'apikey': key,
                'Authorization': `Bearer ${key}`
            }
        })
        const data = await response.json()
        const paths = data.paths || {}
        const rpcs = Object.keys(paths).filter(path => path.startsWith('/rpc/'))
        console.log('Available RPCs:')
        rpcs.forEach(rpc => console.log('  ', rpc))
    } catch (e) {
        console.error('Error:', e)
    }
}

run()
